### ISSUE TITLE
Lint on returning unbounded lifetimes

### ISSUE THREAD

**Issue by Manishearth** [Manishearth/rust-clippy#345]
http://erickt.github.io/blog/2015/09/22/if-you-use-unsafe/

If there is an output lifetime which is not bound to any input lifetimes, for a function containing `unsafe`, the returned lifetime can be anything up to `'static`, and can be used to create pointers which live longer than their pointees.

**Comment by erickt** [comment]
The unsafety I talked about doesn't actually need `unsafe` in this immediate function body. My bug came about due to a combination of raw pointers and `PhantomData`. This pointed then was able to jailbreak out from its lifetime to cause havoc mayhem when used in another `unsafe` block in some other function.

**Comment by llogiq** [comment]
I don't think it needs unsafety at all to wreak havoc – most things have some unsafe things on the code paths. However, if we return a `'static` lifetime, we better have some `const` where it comes from under the rug.

**Comment by Manishearth** [comment]
Yeah, we found a similar situation in Servo that doesn't have unsafe due to module-local unsafety
